PopoverProps Incompatible with New React Typings on v1.6.6 (Bootstrap 4)

### Prerequisites

- [X] I am using the [correct version](https://github.com/react-bootstrap/react-bootstrap#bootstrap-compatibility) of React-Bootstrap for my version of Bootstrap
- [X] I have [searched](https://github.com/react-bootstrap/react-bootstrap/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/react-bootstrap/react-bootstrap/blob/master/CONTRIBUTING.md)

### Describe the bug

Hello,

I'm still on Bootstrap 4 so I'm using 1.6.6, and after upgrading React typings recently I ran into this error coming from React Bootstrap's Popover typings:

```
error TS2430: Interface 'PopoverProps' incorrectly extends interface 'Omit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "ref">'.
  Types of property 'content' are incompatible.
    Type 'boolean | undefined' is not assignable to type 'string | undefined'.
      Type 'boolean' is not assignable to type 'string'.

6 export interface PopoverProps extends React.ComponentPropsWithoutRef<'div'>, BsPrefixPropsWithChildren {
                   ~~~~~~~~~~~~


Found 1 error in node_modules/react-bootstrap/esm/Popover.d.ts:6
```

I think this line just needs to be updated from `boolean` to `string`, but I'm not sure because I don't even see this line on the main branch anymore:

https://github.com/react-bootstrap/react-bootstrap/blob/14add73430e8238b31e8b6af865c037997791414/src/Popover.tsx#L21

### Expected behavior

No errors when compiling types

### To Reproduce

1. Clone the example repo
2. Run `npm install && npx tsc --noEmit`

### Reproducible Example

https://github.com/hueter/react-bootstrap-typings-bug/tree/main

### Screenshots

<img width="1707" alt="Screenshot 2023-04-26 at 8 31 37 PM" src="https://user-images.githubusercontent.com/13444851/234730355-380223bc-9eed-4ac8-9190-fc04676a1c66.png">


### What operating system(s) are you seeing the problem on?

_No response_

### What browser(s) are you seeing the problem on?

_No response_

### What version of React-Bootstrap are you using?

1.6.6

### What version of Bootstrap are you using?

4.6.2

### Additional context

```json
{
  "dependencies": {
    "@types/react": "^18.2.0",
    "@types/react-dom": "^18.2.1",
    "bootstrap": "4.6.2",
    "react": "^18.2.0",
    "react-bootstrap": "1.6.6",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "typescript": "^4.9.5"
  }
}
```

--- a/src/Popover.tsx
+++ b/src/Popover.tsx
@@ -12,7 +12,7 @@ import {
 } from './helpers';
 
 export interface PopoverProps
-  extends React.ComponentPropsWithoutRef<'div'>,
+  extends Omit<React.ComponentPropsWithoutRef<'div'>, 'content'>,
     BsPrefixPropsWithChildren {
   id: string;
   placement?: Placement;
@@ -77,7 +77,7 @@ const defaultProps = {
   placement: 'right',
 };
 
-const Popover: Popover = (React.forwardRef<HTMLDivElement, PopoverProps>(
+const Popover: Popover = React.forwardRef<HTMLDivElement, PopoverProps>(
   (
     {
       bsPrefix,
@@ -114,7 +114,7 @@ const Popover: Popover = (React.forwardRef<HTMLDivElement, PopoverProps>(
       </div>
     );
   },
-) as unknown) as Popover;
+) as unknown as Popover;
 
 Popover.propTypes = propTypes;
 Popover.defaultProps = defaultProps as any;

